src/extension/background-script/actions/nostr/signEventOrPrompt.ts

```diff
@@ -0,0 +1,55 @@
+import { decryptData } from "~/common/lib/crypto";
+import utils from "~/common/lib/utils";
+import { Event } from "~/extension/ln/nostr/types";
+import { MessageSignEvent } from "~/types";
+
+import state from "../../state";
+import { signEvent, validateEvent } from "./helpers";
+
+const signEventOrPrompt = async (message: MessageSignEvent) => {
+  if (!("host" in message.origin)) {
+    console.error("error", message.origin);
+    return;
+  }
+
+  if (!validateEvent(message.args.event)) {
+    console.error("Invalid event");
+    return {
+      error: "Invalid event.",
+    };
+  }
+
+  // Set the message as the user needs to see the event details
+  message.args.message = JSON.stringify(message.args.event);
+
+  try {
+    const response = await utils.openPrompt({
+      ...message,
+      action: "confirmSignMessage",
+    });
+
+    const pw = state.getState().password;
```

Comment:
was wondering if we might want to move the signing (which includes the lookup and decryption of the private key) out of the action.

maybe we can do this similarly as with the connector which is an object we keep in the `state`.
then we don't need to handle the password and keys in different actions

Reply:
generally just because this is sensitive code and data. having this in a single place sounds to me less error prone.

Reply:
moved everything to a Nostr object in state.